docs/frontend-system/building-plugins/06-swappable-components.md

```diff
@@ -0,0 +1,129 @@
+    <div>Your name is {props.name}</div>,
+
+  // This is an optional function that can be used to transform the props of the external component.
+  // If it's not provided, the props of the external component will be passed through unchanged.
+  // Typically this wouldn't be used and you don't need to define this initially,
+  // this is a nice API to evolve the Swappable Component internal API without breaking usages of it in existing plugins.
+  transformProps: (props: { lastName: string; firstName: string }) => ({
+    name: `${props.firstName} ${props.lastName}`,
+  }),
+});
+```
+
+## Using a Swappable Component
+
+Using a Swappable Component is just like using any other React Component. The return of `createSwappableComponent` can be rendered directly in your plugin, or any plugins that will be consuming your `-react` package.
+
+```tsx title="in @internal/plugin-example"
+import { ExampleSwappableComponent } from '@internal/plugin-example-react';
+
+<ExampleSwappableComponent firstName="John" lastName="Doe" />;
+```
+
+## Overriding a Swappable Component
+
+In order to override a Swappable Component, you need to create a `SwappableComponentBlueprint` and install it with the `app` plugin.
+There's two different ways to add extensions to the `app` plugin, both are documented below in an example of overriding the `Progress` Swappable Component.
+
+```tsx title="in packages/app/src/App.tsx"
+import {
+  Progress,
+  SwappableComponentBlueprint,
+  createFrontendModule,
+} from '@backstage/frontend-plugin-api';
+import { MyCustomProgress } from './CustomProgress';
+import { createApp } from '@backstage/frontend-defaults';
+import appPlugin from '@backstage/plugin-app';
+
+const app = createApp({
+  features: [
+    // Override an existing extension by ID provided by the app plugin:
+    appPlugin.withOverrides({
+      extensions: [
+        appPlugin
+          .getExtension('component:app/core.components.progress')
+          .override({
+            params: defineParams =>
+              defineParams({
+                component: Progress,
+                loader: () => MyCustomProgress,
+              }),
+          }),
+      ],
+    }),
+    // OR: Add another extension but with the same component ID:
+    appPlugin.withOverrides({
+      extensions: [
+        SwappableComponentBlueprint.make({
+          name: 'core.components.progress',
+          params: defineParams =>
+            defineParams({
+              component: Progress,
+              loader: () => MyCustomProgress,
+            }),
+        }),
+      ],
+    }),
+    // OR: Use a module for the app plugin:
+    createFrontendModule({
+      pluginId: 'app',
+      extensions: [
+        SwappableComponentBlueprint.make({
+          name: 'core.components.progress',
+          params: defineParams =>
+            defineParams({
+              component: Progress,
+              loader: () => MyCustomProgress,
+            }),
+        }),
+      ],
+    }),
```

Comment:
I'm feeling this is the better one to recommend tbh, because if you're overriding directly via the `appPlugin` instance it's safer to use the `.getExtension` approach. You can't do that if you want to ship it as a module though, which makes this example useful as well. So keep the first example, skip the 2nd, and keep this.